api/v1/search/mongodbsearch_types.go

```diff
@@ -47,6 +49,22 @@ type MongoDBSource struct {
 	Username *string `json:"username,omitempty"`
 }
 
+type Security struct {
+	// +optional
+	TLS TLS `json:"tls"`
+}
+
+type TLS struct {
+	Enabled bool `json:"enabled"`
+	// CertificateKeySecret is a reference to a Secret containing a private key and certificate to use for TLS.
+	// The key and cert are expected to be PEM encoded and available at "tls.key" and "tls.crt".
+	// This is the same format used for the standard "kubernetes.io/tls" Secret type, but no specific type is required.
+	// Alternatively, an entry tls.pem, containing the concatenation of cert and key, can be provided.
```

Comment:
do we normally support tls.pem AND tls.crt in secrets? I thought we're only supporting standard tls secret types so tls.key and tls.crt only. We might have some leftovers in the codebase handling tls.pem-type secrets but we shouldn't support them anymore for new things. Could you verify it's the case here?

Reply:
We’re using the community operator’s tls secret handling behind the scenes which does support the concatenated tls.pem in the field. I copied this comment directly from the Community CRD. I am happy to drop the tls.pem thing altogether, though.

Reply:
Let's drop it and keep it simple, exactly the same as tls-type secrets are defined 👍